src/sources/socket/tcp.rs

```diff
@@ -43,14 +46,14 @@ pub struct TcpConfig {
     /// By default, the [global `log_schema.host_key` option][global_host_key] is used.
     ///
     /// [global_host_key]: https://vector.dev/docs/reference/configuration/global-options/#log_schema.host_key
-    host_key: Option<String>,
+    host_key: Option<OptionalValuePath>,
```

Comment:
`Option<Optional...>` is a little weird, but I don't have much context on this type.